core/common/src/main/java/alluxio/metrics/MetricsSystem.java

```diff
@@ -39,225 +43,241 @@
  * The syntax of the metrics configuration file is:
  * [instance].[sink|source].[name].[options]=[value]
  */
-@NotThreadSafe
-public class MetricsSystem {
+@ThreadSafe
+public final class MetricsSystem {
   private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
 
   // Supported special instance names.
   public static final String MASTER_INSTANCE = "master";
   public static final String WORKER_INSTANCE = "worker";
+  public static final String CLIENT_INSTANCE = "client";
+
+  public static final MetricRegistry METRIC_REGISTRY;
+
+  static {
+    METRIC_REGISTRY = new MetricRegistry();
+    METRIC_REGISTRY.registerAll(new GarbageCollectorMetricSet());
+    METRIC_REGISTRY.registerAll(new MemoryUsageGaugeSet());
+  }
+
+  private static List<Sink> sSinks;
 
   public static final String SINK_REGEX = "^sink\\.(.+)\\.(.+)";
-  public static final String SOURCE_REGEX = "^source\\.(.+)\\.(.+)";
   private static final TimeUnit MINIMAL_POLL_UNIT = TimeUnit.SECONDS;
   private static final int MINIMAL_POLL_PERIOD = 1;
 
-  private String mInstance;
-  private List<Sink> mSinks = new ArrayList<>();
-  private List<Source> mSources = new ArrayList<>();
-  private MetricRegistry mMetricRegistry = new MetricRegistry();
-  private MetricsConfig mMetricsConfig;
-  private boolean mRunning = false;
+  /**
+   * Start sinks specified in the configuration. This is an no-op if the sinks have already been
+   * started.
+   * Note: This has to be called after Alluxio configuration is initialized.
+   */
+  public static void startSinks() {
+    String metricsConfFile = Configuration.get(PropertyKey.METRICS_CONF_FILE);
+    if (metricsConfFile.isEmpty()) {
```

Comment:
(1) is metrics turned on by default?
(2) does it mean to disable metrics system, we need to overwite the `METRICS_CONF_FILE` property to be null? Is this documented any where?